component/component.go

```diff
@@ -0,0 +1,90 @@
+// Package component describes the interfaces which Flow components implement.
+//
+// A Flow component is a distinct piece of business logic that accepts inputs
+// (Arguments) for its configuration and can optionally export a set of outputs
```

Comment:
I think input/output is way more intuitive but it gets confusing for components which output a field which can accept data being pushed to it. That was the reason I went for Terraform's terminology of arguments/exported fields, since "this exported field accepts metrics as input" is easier to follow IMO.